piecrust-uplink/src/abi/state.rs

```diff
@@ -292,6 +294,37 @@ pub fn owner<const N: usize>(contract: ContractId) -> Option<[u8; N]> {
     }
 }
 
+/// Returns given contract's free limit, if the contract exists and if it
+/// has a free limit set.
+pub fn free_limit(contract: ContractId) -> Option<u64> {
+    let contract_id_ptr = contract.as_bytes().as_ptr();
+    unsafe {
+        match ext::free_limit(contract_id_ptr) as usize {
+            0 => None,
+            arg_pos => with_arg_buf(|buf, _| {
+                let ret = archived_root::<Option<u64>>(&buf[..arg_pos]);
+                ret.deserialize(&mut Infallible).expect("Infallible")
```

Comment:
I would like to prevent the addition of `rkyv` into the ABI if possible. This is due to `rkyv` being fairly Rust-specific. I would suggest coming up with a custom-yet-sane™ serialization for this integer. Can we use the return of the `extern` instead?

Reply:
sounds good, I'll do it

Reply:
done